int32    mode
Joints[] joints

